flintrock/core.py

```diff
@@ -482,6 +483,58 @@ def _run_asynchronously(*, partial_func: functools.partial, hosts: list):
         loop.close()
 
 
+def get_java_major_version(client: paramiko.client.SSHClient):
+    possible_cmds = [
+        "$JAVA_HOME/bin/java -version",
+        "java -version"
+    ]
+
+    for command in possible_cmds:
+        try:
+            output = ssh_check_output(
```

Comment:
Hmm, I think this will always return an empty byte string because `java -version` spits its output on `stderr`, whereas `ssh_check_output()` only returns `stdout`.

Ultimately, I want to change `ssh_check_output()` so that it cleanly returns both `stdout` and `stderr`, but in the meantime you can fix this by updating your `possible_cmds` to redirect `stderr` to `stdout`.

```sh
$ORIGINAL_COMMAND 2>&1
```